io.openems.edge.evcs.dezony/src/io/openems/edge/evcs/dezony/DezonyApi.java

```diff
@@ -0,0 +1,154 @@
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.exceptions.OpenemsException;
+import io.openems.common.utils.JsonUtils;
+
+/**
+ * Implements the dezony REST-Api.
+ *
+ */
+public class DezonyApi {
+	private final String baseUrl;
+	private final DezonyImpl dezonyImpl;
+
+	public DezonyApi(String ip, int port, DezonyImpl dezonyImpl) {
+		this.baseUrl = "http://" + ip + ":" + port;
+		this.dezonyImpl = dezonyImpl;
+	}
+	
+	public JsonElement getState() throws OpenemsNamedException {
+		return this.sendGetRequest("/api/v1/state");
+	}
+	
+	public JsonElement getLastMetric() throws OpenemsNamedException {
+		return this.sendGetRequest("/api/v1/metrics/last");
+	}
+	
+	public JsonElement enalbeCharing() throws OpenemsNamedException {
+		return this.sendPostRequest("/api/v1/charging/unlock");
+	}
+	
+	public boolean disableCharing() throws OpenemsNamedException {
+		final var resultLimit = this.sendPostRequest("/api/v1/charging/lock");;
+		final var result = JsonUtils.getAsOptionalBoolean(resultLimit, "charging_is_locked");
+		
+		return result.orElse(false).equals(true);
+	}
+	
+	public Optional<String> setCurrent(int current) throws OpenemsNamedException {
+		final var resultLimit = this.sendPostRequest("/api/v1/charging/current?value=" + current);
+		
+		return JsonUtils.getAsOptionalString(resultLimit, "charging_current");
+	}
+
+	/**
+	 * Sends a get request.
+	 *
+	 * @param endpoint the REST Api endpoint
+	 * @return a JsonObject or JsonArray
+	 * @throws OpenemsNamedException on error
+	 */
+	public JsonElement sendGetRequest(String endpoint) throws OpenemsNamedException {
+	    try {
+	        URL url = new URL(this.baseUrl + endpoint);
+	        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+
+	        con.setRequestMethod("GET");
+	        con.setConnectTimeout(5000);
+	        con.setReadTimeout(5000);
+
+	        try (BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()))) {
+	            StringBuilder content = new StringBuilder();
+	            String line;
+
+	            while ((line = in.readLine()) != null) {
+	                content.append(line);
+	                content.append(System.lineSeparator());
+	            }
+
+	            String body = content.toString();
+
+	            // Get response code
+	            int status = con.getResponseCode();
+
+	            if (status >= 300) {
+	                throw new OpenemsException("Error while reading from dezony API. Response code: " + status + ". " + body);
+	            }
+
+	            return JsonUtils.parseToJsonObject(body);
+	        }
+	    } catch (IOException e) {
+	        // Log the error
+	        e.printStackTrace();
+	    }
+
+	    // If there was an error, set the communication failed flag to true and return null
+	    this.dezonyImpl._setChargingstationCommunicationFailed(true);
```

Comment:
The value is not reset to false again when the connection is re-established.
Maybe you could do it like in the sendPostRequest method.